x-pack/plugins/security_solution/public/siem_migrations/rules/components/rule_details_flyout/index.tsx

```diff
@@ -76,38 +74,38 @@ export const MigrationRuleDetailsFlyout: React.FC<MigrationRuleDetailsFlyoutProp
   ({
     ruleActions,
     ruleMigration,
+    matchedPrebuiltRule,
     size = 'm',
     extraTabs = [],
     closeFlyout,
   }: MigrationRuleDetailsFlyoutProps) => {
     const { expandedOverviewSections, toggleOverviewSection } = useOverviewTabSections();
 
     const rule: RuleResponse = useMemo(() => {
-      const esqlLanguage = ruleMigration.elastic_rule?.query_language ?? 'esql';
+      if (matchedPrebuiltRule) {
+        return matchedPrebuiltRule;
+      }
       return {
-        type: esqlLanguage,
-        language: esqlLanguage,
-        name: ruleMigration.elastic_rule?.title,
-        description: ruleMigration.elastic_rule?.description,
-        query: ruleMigration.elastic_rule?.query,
-
-        ...DEFAULT_TRANSLATION_FIELDS,
-        severity:
-          (ruleMigration.elastic_rule?.severity as Severity) ?? DEFAULT_TRANSLATION_SEVERITY,
+        ...(isMigrationCustomRule(ruleMigration.elastic_rule)
+          ? convertMigrationCustomRuleToSecurityRulePayload(ruleMigration.elastic_rule)
+          : {}),
       } as RuleResponse; // TODO: we need to adjust RuleOverviewTab to allow partial RuleResponse as a parameter
-    }, [ruleMigration]);
+    }, [matchedPrebuiltRule, ruleMigration]);
```

Comment:
Is it possible this returns `{} as RuleResponse`?

Reply:
Not really. I will re-arrange this. It is possible that matched prebuilt rule is undefined though, and we either need to remove overview tab or show some message instead. This should not happen, but since that parameter is optional we need to account for that possibility. Another option will be to throw an exception beforehand if we end up in such a situation or prevent user from opening the flyout at all.